srcts/src/shiny/render.ts

```diff
@@ -220,9 +243,24 @@ function renderDependency(dep_: HtmlDep) {
       script.setAttribute(attr, val ? val : "");
     });
 
-    $head.append(script);
+    const p = new Promise((resolve) => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      script.onload = (e: Event) => {
+        resolve(null);
+      };
+    });
+
+    scriptPromises.push(p);
+    scriptElements.push(script);
   });
 
+  // Append the script elements all at once, so that we're sure they'll load in
+  // order. (We didn't append them individually in the `forEach()` above,
+  // because we're not sure that the browser will load them in order if done
+  // that way.)
+  document.head.append(...scriptElements);
+  await Promise.allSettled(scriptPromises);
```

Comment:
Why `Promise.allSettled()` and not `Promise.all()`? (We are not using the result.)

Reply:
I still don't see why the promise is needed as long as the scripts are loaded in order.

If they're loaded in order, then the whole async/await changes can be removed.

Reply:
> Why `Promise.allSettled()` and not `Promise.all()`? (We are not using the result.)

The `.allSettled()` is there in case there's an error loading any script tags. If it used `.all()`, that would require all the loads to be successful before moving on.

Reply:
The `await` is necessary because we don't want to continue on this code path until all the scripts have run.

For example, we don't want run `bindInput` to initialize a slider element before the slider JS files have loaded.